client/api/src/backend.rs

```diff
@@ -303,32 +303,44 @@ pub trait AuxStore {
 }
 
 /// An `Iterator` that iterates keys in a given block under a prefix.
-pub struct KeyIterator<'a, State, Block> {
+pub struct KeyIterator<State, Block> {
 	state: State,
 	child_storage: Option<ChildInfo>,
-	prefix: Option<&'a StorageKey>,
+	prefix: Option<StorageKey>,
 	current_key: Vec<u8>,
 	_phantom: PhantomData<Block>,
 }
 
-impl<'a, State, Block> KeyIterator<'a, State, Block> {
+impl<State, Block> KeyIterator<State, Block> {
 	/// create a KeyIterator instance
-	pub fn new(state: State, prefix: Option<&'a StorageKey>, current_key: Vec<u8>) -> Self {
-		Self { state, child_storage: None, prefix, current_key, _phantom: PhantomData }
+	pub fn new(state: State, prefix: Option<&StorageKey>, current_key: Vec<u8>) -> Self {
```

Comment:
would be nice to be explicit here and avoid to clone it inside this function after your changes.

```suggestion
	pub fn new(state: State, prefix: Option<StorageKey>, current_key: Vec<u8>) -> Self {
```